src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -393,22 +394,34 @@ class Holder {
 
     private void emitConstant(Class<?> javaType, String constantName, Object value, Declaration declaration) {
         incrAlign();
-        appendLines(STR."""
-            private static final \{javaType.getSimpleName()} \{constantName} = \{constantValue(javaType, value)};
-
-            """);
-        emitDocComment(declaration);
-        appendLines(STR."""
-            public static \{javaType.getSimpleName()} \{constantName}() {
-                return \{constantName};
-            }
-            """);
+        boolean useHolderClass = value instanceof String;
```

Comment:
maybe `isString` would be a better name

Reply:
I'll just drop the intermediate variable